### ISSUE TITLE
Cannot use install script INSTALL_K3S_PR to install recent PR builds

### ISSUE THREAD

**Issue by brandond** [k3s-io/k3s#11382]
This appears to be due to recent changes to CI causing there to be more than 30 checks in the result, causing some to be omitted from the api query:
```console
brandond@dev01:~/go/src/github.com/k3s-io/k3s$ curl -s -H 'Authorization: Bearer gho_XXX' https://api.github.com/repos/k3s-io/k3s/commits/d39ab65ca69ab7b7c6ea3825e1257e7812e57168/check-runs | jq '.check_runs[] | .name'
"codecov/project/pkg"
"Docker Tests (upgrade)"
"Docker Tests In GO (upgrade)"
"Docker Tests (lazypull)"
"Docker Tests In GO (skew)"
"E2E Tests (wasm)"
"Docker Tests In GO (lazypull)"
"E2E Tests (embeddedmirror)"
"Docker Tests In GO (etcd)"
"E2E Tests (privateregistry)"
"Docker Tests In GO (cacerts)"
"E2E Tests (externalip)"
"Docker Tests (compat)"
"Docker Tests In GO (bootstraptoken)"
"E2E Tests (btrfs)"
"Docker Tests (cacerts)"
"E2E Tests (s3)"
"Docker Tests (bootstraptoken)"
"Docker Tests In GO (basics)"
"Docker Tests (basics)"
"E2E Tests (startup)"
"E2E Tests (autoimport)"
"Integration Tests (flannelnone)"
"Integration Tests (secretsencryption)"
"Integration Tests (longhorn)"
"Integration Tests (kubeflags)"
"Integration Tests (etcdsnapshot)"
"Integration Tests (custometcdargs)"
"Integration Tests (startup)"
"Integration Tests (localstorage)"
```

Instead of listing all checks and then filtering them by name with `jq`, we should ask the server to filter them for us via the `check_name` query parameter.

**Comment by VestigeJ** [comment]
This is an internal facing issue only for internal team members

TLDR 
```
$ export GITHUB_TOKEN=your_very_own_secret_potato

//installing old PR
$ curl -sL get.k3s.io | INSTALL_K3S_PR=10973 sh -s -
[INFO]  Using PR 10973 as release
[INFO]  Downloading hash https://api.github.com/repos/k3s-io/k3s/actions/artifacts/2264136161/zip
[INFO]  Verifying binary download
[INFO]  Installing k3s to /usr/local/bin/k3s
[INFO]  Skipping installation of SELinux RPM
[INFO]  Creating /usr/local/bin/kubectl symlink to k3s
[INFO]  Creating /usr/local/bin/crictl symlink to k3s
[INFO]  Creating /usr/local/bin/ctr symlink to k3s
[INFO]  Creating killall script /usr/local/bin/k3s-killall.sh
[INFO]  Creating uninstall script /usr/local/bin/k3s-uninstall.sh
[INFO]  env: Creating environment file /etc/systemd/system/k3s.service.env
[INFO]  systemd: Creating service file /etc/systemd/system/k3s.service
[INFO]  systemd: Enabling k3s unit
Created symlink /etc/systemd/system/multi-user.target.wants/k3s.service → /etc/systemd/system/k3s.service.
[INFO]  systemd: Starting k3s

//installing new PR
$ curl -sL get.k3s.io | INSTALL_K3S_PR=11448 sh -s -
[INFO]  Using PR 11448 as release
[INFO]  Downloading hash https://api.github.com/repos/k3s-io/k3s/actions/artifacts/2307242461/zip
[INFO]  Verifying binary download
[INFO]  Installing k3s to /usr/local/bin/k3s
[INFO]  Skipping installation of SELinux RPM
[INFO]  Creating /usr/local/bin/kubectl symlink to k3s
[INFO]  Creating /usr/local/bin/crictl symlink to k3s
[INFO]  Creating /usr/local/bin/ctr symlink to k3s
[INFO]  Creating killall script /usr/local/bin/k3s-killall.sh
[INFO]  Creating uninstall script /usr/local/bin/k3s-uninstall.sh
[INFO]  env: Creating environment file /etc/systemd/system/k3s.service.env
[INFO]  systemd: Creating service file /etc/systemd/system/k3s.service
[INFO]  systemd: Enabling k3s unit
Created symlink /etc/systemd/system/multi-user.target.wants/k3s.service → /etc/systemd/system/k3s.service.
[INFO]  systemd: Starting k3s
```
